apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/functions/NotificationFunctions.java

```diff
@@ -115,4 +128,66 @@ public HttpResponseMessage countNotifications(
                 .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
                 .build();
     }
+
+    /**
+     * This HTTP-triggered function invokes an orchestration to resend notifications for a given range of dates and filters
+     */
+    @FunctionName("ResendNotifications")
+    public HttpResponseMessage resendNotifications(
+            @HttpTrigger(name = "req", methods = {HttpMethod.POST}, authLevel = AuthorizationLevel.FUNCTION) HttpRequestMessage<Optional<String>> request,
+            @DurableClientInput(name = "durableContext") DurableClientContext durableContext,
+            final ExecutionContext context) throws JsonProcessingException {
+        context.getLogger().info("resendNotifications trigger processed a request");
+
+        ResendNotificationsFilters filters = getResendNotificationsFilters(request);
+        try {
+            checkResendNotificationsFilters(filters);
+        } catch (IllegalArgumentException e) {
+            return request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+                    .body(e.getMessage())
+                    .build();
+        }
+
+        String filtersInJson = objectMapper.writeValueAsString(filters);
+        DurableTaskClient client = durableContext.getClient();
+        String instanceId = client.scheduleNewOrchestrationInstance("NotificationsSender", filtersInJson);
+        context.getLogger().info(String.format("%s %s", CREATED_NEW_RESEND_NOTIFICATIONS_ORCHESTRATION_WITH_INSTANCE_ID_MSG, instanceId));
+
+        return durableContext.createCheckStatusResponse(request, instanceId);
+    }
+
+    /**
+     * This function is the orchestrator that manages the resend notifications process, it just calls the activity
+     * that performs the actual work
+     */
+    @FunctionName("NotificationsSender")
+    public void notificationsSenderOrchestrator(
+            @DurableOrchestrationTrigger(name = "taskOrchestrationContext") TaskOrchestrationContext ctx,
+            ExecutionContext functionContext) {
+        String filtersString = ctx.getInput(String.class);
+        functionContext.getLogger().info("Resend notifications orchestration started with input: " + filtersString);
+        do {
+            filtersString = ctx.callActivity(RESEND_NOTIFICATIONS_ACTIVITY, filtersString, String.class).await();
+        } while (filtersString != null);
```

Comment:
this entire business logic could be more tricky....iterate over durable functions using do-while may more complicated to maintain and to debug in case of errors

Reply:
my doubt is about do-while, it could be an anti-pattern as explained in [this article](https://karl-henrik.se/posts/Durable-Functions-Loop-Anti-Pattern/)....In addition, orchestrator functions call only [deterministic APIs](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-code-constraints?tabs=csharp#using-deterministic-apis), this business logic does not permit deterministic iteration